ClosedXML/Excel/CalcEngine/XLCalcEngine.cs

```diff
@@ -95,11 +95,36 @@ public override object GetExternalObject(string identifier)
                 }
             }
             else if (_ws != null)
+            {
+                if (TryGetNamedRange(identifier, _ws, out IXLNamedRange namedRange))
+                {
+                    var references = (namedRange as XLNamedRange).RangeList.Select(r =>
+                        XLHelper.IsValidRangeAddress(r) 
+                            ? GetCellRangeReference(_ws.Workbook.Range(r))
+                            : new XLCalcEngine(_ws).Evaluate(r.ToString())
+                        );
+                    if (references.Count() == 1)
+                        return references.Single();
+                    return references;
+                }
+
                 return GetCellRangeReference(_ws.Range(identifier));
+            }
             else
                 return identifier;
         }
 
+        private bool TryGetNamedRange(string identifier, IXLWorksheet worksheet, out IXLNamedRange namedRange)
+        {
+            Func<string, string, bool> equals = (a, b) =>
```

Comment:
`bool equals(string a, string b) => string.Equals(a, b, StringComparison.InvariantCultureIgnoreCase);` 
simpler?

Reply:
Hmm, or it's kind of the same complexity.